app/fetch-did-web-doc.py

```diff
@@ -120,10 +143,13 @@ def request(url, data=None, headers=None):
 
 
 def run(url, nonce, callback_url: str, unattested: bool):
-    if unattested:
-        result = fetch_unattested(url, nonce)
-    else:
-        result = fetch_attested(url, nonce)
+    fetch = fetch_unattested if unattested else fetch_attested
+    retry = True
+    tries = 0
+    while retry and tries < FETCH_RETRIES:
+        result = fetch(url, nonce)
+        retry = refetchable(result)
+        tries += 1
```

Comment:
`fetch_unattested` uses `request`, which already has its own retry loop. IMO the retry loop here should be pushed into the fetch_attested function.

Reply:
I've updated this to only retry the attested fetch - I do think it's more clear to keep this level of retry here given we are inspecting the output of attested fetch to decide whether to repeat it, i.e. attested-fetch is completing and we are then repeating it. 
The attested-fetch function already has retry logic to deal with network issues, so potentially where attested-fetch is attesting to a curl error we could skip the higher level retries.

Reply:
> The attested-fetch function already has retry logic to deal with network issues, so potentially where attested-fetch is attesting to a curl error we could skip the higher level retries.

I think that's not accurate as network issues also lead to an attested fetch result (where `error` is present). Only when the enclave crashes for some reason would the existing retry logic in `fetch_attested` kick in now. I agree with @plietar that the retry loop should be inside `fetch_attested` but we can also do that in a follow-up PR.